src/main/java/com/app/bookstore/dto/CreateBookRequestDto.java

```diff
@@ -1,13 +1,20 @@
 package com.app.bookstore.dto;
 
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Positive;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
     private String title;

```

Comment:
I would also validate maximum size of text fields, to not exceed its length in DB and not get an exception. (And minimum size too)